platform/kobo/koreader.sh

```diff
@@ -611,10 +611,20 @@ if [ ${RETURN_VALUE} -ne ${KO_RC_HALT} ]; then
         # if we were called from advboot then we must reboot to go to the menu
         # NOTE: This is actually achieved by checking if KSM or a KSM-related script is running:
         #       This might lead to false-positives if you use neither KSM nor advboot to launch KOReader *without nickel running*.
-        if ! pgrep -f kbmenu >/dev/null 2>&1; then
+        if ! pkill -0 kbmenu; then
             /sbin/reboot
         fi
     fi
+else
+    if [ "${VIA_NICKEL}" = "false" ]; then
+        if pkill -0 kbmenu; then
+            # If we were started by KSM and requested an exit, do *NOT* exit the script so as not to re-enter KSM at all,
+            # to make sure the device powers off with our own ScreenSaver displayed.
+            while :; do
+                sleep 2073600
+            done
```

Comment:
I.e.,
```suggestion
            sleep 10
```

Reply:
This risks failing by exiting too early, or failing outright if interrupted by a signal ;).

Reply:
The other approach would to just fudge the return code to 0, but that would still display KSM over our screensaver (I imagine?).

Reply:
The observed behavior is:

* KOReader shutdown screen
* KOReader shuts down and KSM takes over for a second

I don't see how adding 10 seconds risks exiting too early.

A signal will also shut down KSM.

Reply:
Forgot to add, I confirmed that it works as I said. I'm not talking in hypotheticals.